md.py

```diff
@@ -184,20 +190,21 @@ def setup_from_args(self, args=None):
         args = parser.parse_args(args)
 
         if isinstance(args.devs, str):
-            args.devs = args.devs.split()
+            if args.devs == "":
+                args.devs = None
+            else:
+                args.devs = args.devs.split()
 
-        if args.devs:
-            args.disks = None
-            args.loop_disks = None
-        if args.loop_disks:
-            args.disks = None
+        if args.disk_type == 'default':
+            args.disk_type = 'ram' if args.devs is None else 'dev'
 
-        if not args.devs and not args.loop_disks and not args.disks:
+        if not args.disks:
```

Comment:
If `args.disks == 0 and args.disk_type == 'dev'`, then I think it might be better to set `args.disks = len(args.devs)`

Reply:
There are lots of tests that use extra disks for things, we don't want to default to len(args.devs) ever or that will break things

Reply:
when running batch tests you set the DEVS env once for all tests so it has to contain the maximum number of disks for any test. Some tests use one of the unused disks as a jornal dev, and test3 uses one as a verification disk. MDInstance has no way of knowing how these will be used, and since the default ndisks is used for both journaling and test3 we cannot use all of the devs provided by args.devs.

Reply:
Oh wait, on closer inspection args.disks is basically just a 0 check, so the default args is mute point since it is 3 which will not affect this.  Did you mean if someone specifies --disks 0 it should pull the len from devs or use the default 3 if none are present? I'm not completely opposed to that, but it does make it so ram/loop tests can run with journaling or test3 verification enabled and DEVS tests cannot.

Reply:
How about just renaming DEVS to AVAILABLE_DEVS? That way it's more clear they won't all be used, and doesn't change the way tests work when running with or without explicit devs.

Reply:
I would say AVAILABLE_DEVS is too long for an environment variable and not really necessary.

Reply:
Ok, Leave all this as close to the original as possible (drop args.disks default).  I don't actually object to changing the way things work, but this pr is about getting all tests to work with a DEVS=... env arg. Changes to test behavious can be done in a separate pr.